coderd/database/migrations/create_migration.sh

```diff
@@ -7,6 +7,17 @@
 
 set -euo pipefail
 
+cat <<EOF
+
+WARNING: Migrations now all run in a single transaction. This makes upgrades
+safer, but means that 'ALTER TYPE resource_type ADD VALUE' cannot be used if the
+enum value needs to be referenced in another migration.
```

Comment:
Documentation alone is not sufficient IMO (hence suggesting code-gen). This is the type of issue that might surface a week, month, or year later when a migration using the earlier type is added. And only _some_ customers will hit it (those that rarely update).

Reply:
This isn't really a bug that customers will run into, it'll always be caught by tests. The issue only pops up when you attempt to use an enum value added by `ADD VALUE` in a later migration. Otherwise, these are fine to use.

Reply:
I don't think we can _always_ catch this, esp. without type-gen, linting, etc. One scenario I can think of is:

1. Customer A and B run v2.0.0
2. We add a value (`foo(bar)`) in v2.0.1, but we forget the type dance because there's nothing to catch it when the value is unused
3. Customer B updates to v2.0.1
4. We release v2.0.2 and add a migration that modifies rows and change `foo(baz) => foo(bar)`, we don't catch it because we don't have fixtures for everything and the relevant tables are empty
5. Customer A and B update to v2.0.2
6. Customer A runs into an error

Reply:
The migration tests will catch this since it doesn't matter if the tables are empty. Any reference to the new enum will fail. I ran into this exact scenario while writing this.

https://github.com/coder/coder/blob/main/coderd/database/migrations/000103_add_apikey_name.up.sql does an `UPDATE` using the `login_type(token)` enum added using `ADD VALUE` in an earlier migration. The tests are able to catch this without fixtures since it doesn't matter if any rows are in the table. It still has to convert `'token'` to the proper enum, which causes the error.

Reply:
Oh ok, that's good then. I wasn't aware that postgres would catch it for an empty table. Thanks for clarifying. 👍🏻